### PR TITLE
RavenDB-20184 : Sharding - Restore - Ensure we can restore to a given point

### PR DESCRIPTION
**Pull request by aviv86**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-20184

### Additional description

- refactor `RestorePointBase` and its subclasses + minor modifications to `/admin/restore/points` endpoint
- add `LastFileNameToRestore` property to `SingleShardRestoreSetting`,  which will be used to restore each shard up to a specific backup file
- tests

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.

### Testing by Contributor

- Tests have been added that prove the fix is effective or that the feature works

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- It requires further work in the Studio

### CODE REVIEW

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestorePointsBase.cs [hunk 1]
```diff
@@ -95,7 +95,32 @@ public class ParsedBackupFolderName
...
+        protected Task<RestorePoints> FetchRestorePointsForPath(string path, bool assertLegacyBackups)
+        {
+            return FetchRestorePointsForPaths(new[] { path }, assertLegacyBackups);
+        }
+
+        protected async Task<RestorePoints> FetchRestorePointsForPaths(IEnumerable<string> paths, bool assertLegacyBackups)
+        {
+            foreach (var path in paths)
+            {
+                await FetchRestorePoints(path, assertLegacyBackups);
```
Comment (karmeli87) [inline src/Raven.Server/Documents/PeriodicBackup/Restore/RestorePointsBase.cs]: Does it make sense to make this in parallel?

**Review by karmeli87 — COMMENTED** [review]
Please check failing tests

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestorePointsBase.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/PeriodicBackup/Restore/RestorePointsBase.cs]: Not a hot path IMO, not worth it

**Review by grisha-kotler — APPROVED** [review]

**Comment by aviv86** [comment]
@kalczur will open a new PR based on this branch with a working UI

**Pull request closed without merging**